R/binary_classification.R

```diff
@@ -138,3 +140,244 @@ fbeta_score <- function(actual, predicted, beta = 1) {
     rec <- recall(actual, predicted)
     return((1 + beta^2) * prec * rec / ((beta^2 * prec) + rec))
 }
+
+#' Binary confusion matrix
+#'
+#' \code{confusion_matrix} Calculates a binary classification confusion matrix,
+#' comparing the predicted with the actual values for the classes.
+#' Assumes that 1 is used for the positive class and 0 for the
+#' negative class.
+#'
+#' Returns a \code{data.frame} with columns corresponding to the
+#' number of True Positives (\code{tp}), False Positives (\code{fp}),
+#' True Negatives (\code{tn}), and False Negatives (\code{fn})
+#'
+#' @inheritParams params_binary
+#' @seealso \code{\link{sensitivity}} \code{\link{specificity}}
+#' @examples
+#' actual <- c(1, 1, 1, 0, 0, 0)
+#' predicted <- c(1, 0, 1, 1, 1, 1)
+#' confusion_matrix(actual, predicted)
+confusion_matrix <- function(actual, predicted) {
+    binvals <- c(0, 1)
+    b_actual <- unique(actual)
+    b_predicted <- unique(predicted)
+
+    # ideally "actual" should be a combination of 0s and 1s,
+    # but could be all 0s or all 1s as degenerate cases
+    if (!(
+        setequal(binvals, b_actual) |
+        setequal(c(0), b_actual) |
+        setequal(c(1), b_actual)
```

Comment:
You don't need `b_actual` nor the three calls to `setequal()`, but instead `!all(actual %in% 0:1)` should do.

Reply:
Wanted to write so you know I am still interested in helping here. Sorry for not been responsive for quite some time, got a bit of health issues that are resolving, and have not had time to look at the code and suggestions. Hopefully by the end of this week. 

A quick glance seems like @mllg suggestions are on point. As for input validation, it will be a "good thing" (TM), but needs some thought I think, in particular if it involves breaking bc.

Perhaps the idea of putting this into a side branch will be better, so things can be merged to the main branch once they are in their final form.